### DISCUSSION TITLE
[Serenity] Active menu item and visible sub menu item

### DISCUSSION THREAD

**Discussion by dmainardi** [primefaces/community#164] [PrimeFaces Templates]
Hello Primefaces maintainers,
I found two strange behaviors in Serenity Menu

1. When I use [ExtensionlessURLs](https://showcase.omnifaces.org/facesviews/ExtensionlessURLs) I cannot see the active menuItem.
I debugged the application (via browser debugging since it is JavaScript) and I found that the function **restoreMenuState** (315 row) tries to match the href with the pathname; but, since it is extensionless, it does not match (for example: '/foo/bar/dashboard' vs '/foo/bar/dashboard.xhtml').
If I disable ExtensionlessURLs everything works as expected.
Is it possible to use ExtensionlessURLs and active-menu-item? I remember that in previous version (v 3, maybe v 2) it worked fine with and without the xhtml extension.

2. If I click on a submenu link (hierarchy) it does not show the menuitem, because it closes the first (root?) submenu. As before, in prevoiuos version of Serenity it worked.

Can you help me?
Thank you very much.

Davide

**Comment by mertsincan** [comment]
Interesting! Could you please try it with the latest Serenity version? 

Best Regards,

**Comment by tandraschko** [comment]
- "apollo"
          - "atlantis"
          - "avalon"
          - "diamond"
          - "freya"
          - "poseidon"
          - "ultima"
          - "verona"
